src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -4016,6 +4020,8 @@ private void checkSpecialCaller(Class<?> specialCaller, Class<?> refc) throws Il
         private boolean restrictProtectedReceiver(MemberName method) {
             // The accessing class only has the right to use a protected member
             // on itself or a subclass.  Enforce that restriction, from JVMS 5.4.4, etc.
+            // Since array have access hacks, need to guard against array clone() before
```

Comment:
I suggest to move this new comment to `getDirectMethodCommon`, the relevant caller of this method.  See suggested wordings below.